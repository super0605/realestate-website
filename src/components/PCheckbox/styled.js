import styled from "styled-components";
import { Checkbox } from "antd";

export const CustomCheckbox = styled(Checkbox)`
  .ant-checkbox-inner {
    border-width: 2px;
    border-color: ${({ theme }) => theme.colors.muted};
    height: ${({ height }) => height};
    width: ${({ width }) => width};
  }
  .ant-checkbox-inner:focus {
    border-width: 2px;
    border-color: ${({ theme }) => theme.colors.muted};
    outline: ${({ theme }) => theme.colors.primary};
  }
  .ant-checkbox-input:focus + .ant-checkbox-inner {
    border-color: ${({ theme }) => theme.colors.muted};
  }
  .ant-checkbox-checked {
    .ant-checkbox-inner {
      background-color: ${({ theme }) => theme.colors.primary};
      border-color: ${({ theme }) => theme.colors.primary};
    }
    .ant-checkbox-input:focus + .ant-checkbox-inner {
      border-color: ${({ theme }) => theme.colors.primary};
    }
  }
  .ant-checkbox-checked::after {
    border: 1px solid ${({ theme }) => theme.colors.primary};
  }
  &.ant-checkbox-wrapper:hover {
    .ant-checkbox-inner {
      border-color: ${({ theme }) => theme.colors.primary};
    }
  }
  &.ant-checkbox-wrapper:focus {
    .ant-checkbox-inner {
      border-color: ${({ theme }) => theme.colors.muted};
    }
  }
`;
