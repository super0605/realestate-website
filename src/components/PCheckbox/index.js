import React from "react";
import PropTypes from "prop-types";
import { CustomCheckbox } from "./styled";

const PCheckbox = ({ height, width, ...rest }) => {
  return <CustomCheckbox height={height} width={width} {...rest} />;
};

PCheckbox.propTypes = {
  width: PropTypes.string,
  height: PropTypes.string,
};

PCheckbox.defaultProps = {
  width: "14px",
  height: "14px",
};

export default PCheckbox;
