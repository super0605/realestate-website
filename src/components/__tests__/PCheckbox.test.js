import * as React from "react";
import { mount, shallow } from "enzyme";
import PCheckbox from "../PCheckbox";

describe("PCheckbox Component", () => {
  let props;
  let mountedPCheckbox;
  let wrapper;

  const renderWrapper = (bShallow) => {
    if (!mountedPCheckbox) {
      mountedPCheckbox = bShallow
        ? shallow(<PCheckbox {...props} />)
        : mount(<PCheckbox {...props} />);
    }
    return mountedPCheckbox;
  };

  beforeEach(() => {
    wrapper = renderWrapper(true);
  });

  it("should render without problems", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
