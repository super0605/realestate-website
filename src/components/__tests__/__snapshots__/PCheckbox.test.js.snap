// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PCheckbox Component should render without problems 1`] = `
<Styled(Checkbox)
  height="14px"
  width="14px"
/>
`;
